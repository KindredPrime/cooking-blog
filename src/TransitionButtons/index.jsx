import PropTypes from 'prop-types';

function TransitionButtons(props) {
  const { isAtTheBeginning, isAtTheEnd, handlePrevClick, handleNextClick } = props;

  return (
    <div className="TransitionButtons">
      <button
        type="button"
        className="TransitionButtons__prev"
        disabled={isAtTheBeginning}
        onClick={handlePrevClick}
      >
        Previous
      </button>
      <button
        type="button"
        className="TransitionButtons__next"
        disabled={isAtTheEnd}
        onClick={handleNextClick}
      >
        Next
      </button>
    </div>
  );
}

TransitionButtons.propTypes = {
  isAtTheBeginning: PropTypes.bool.isRequired,
  isAtTheEnd: PropTypes.bool.isRequired,
  handlePrevClick: PropTypes.func.isRequired,
  handleNextClick: PropTypes.func.isRequired
};

export default TransitionButtons;