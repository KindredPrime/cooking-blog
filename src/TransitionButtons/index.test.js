import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import TransitionButtons from './index';

describe('TransitionButtons Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <TransitionButtons
        isAtTheBeginning={true}
        isAtTheEnd={true}
        handlePrevClick={() => {}}
        handleNextClick={() => {}}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected, at the beginning', () => {
    render(
      <TransitionButtons
        isAtTheBeginning={true}
        isAtTheEnd={false}
        handlePrevClick={() => {}}
        handleNextClick={() => {}}
      />
    );

    expect(document.body).toMatchSnapshot();
  });

  it('renders the UI as expected, in the middle', () => {
    render(
      <TransitionButtons
        isAtTheBeginning={false}
        isAtTheEnd={false}
        handlePrevClick={() => {}}
        handleNextClick={() => {}}
      />
    );

    expect(document.body).toMatchSnapshot();
  });

  it('renders the UI as expected, at the end', () => {
    render(
      <TransitionButtons
        isAtTheBeginning={false}
        isAtTheEnd={true}
        handlePrevClick={() => {}}
        handleNextClick={() => {}}
      />
    );

    expect(document.body).toMatchSnapshot();
  });
});