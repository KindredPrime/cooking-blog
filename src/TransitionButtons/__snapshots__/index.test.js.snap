// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionButtons Component renders the UI as expected, at the beginning 1`] = `
<body>
  <div>
    <div
      class="TransitionButtons"
    >
      <button
        class="TransitionButtons__prev"
        disabled=""
        type="button"
      >
        Previous
      </button>
      <button
        class="TransitionButtons__next"
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</body>
`;

exports[`TransitionButtons Component renders the UI as expected, at the end 1`] = `
<body>
  <div>
    <div
      class="TransitionButtons"
    >
      <button
        class="TransitionButtons__prev"
        type="button"
      >
        Previous
      </button>
      <button
        class="TransitionButtons__next"
        disabled=""
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</body>
`;

exports[`TransitionButtons Component renders the UI as expected, in the middle 1`] = `
<body>
  <div>
    <div
      class="TransitionButtons"
    >
      <button
        class="TransitionButtons__prev"
        type="button"
      >
        Previous
      </button>
      <button
        class="TransitionButtons__next"
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</body>
`;
